### PR TITLE
Combined recommendedSizes and regex fixes

### PR DESCRIPTION
**Pull request by QuasarSE**
Combination pull request with both of:
1. Change for VM SizeSelector Control to reduce the minimum size of the recommendedSizes array from 3 to 1.
2. Expose the regex and validationMessage constraints on the UserNameTextBox control.

### CODE REVIEW

**Comment by QuasarSE** [comment]
Note that this pr is a superset of #294.   If this one is applied, no need to apply #294.

**Comment by TianoMS** [comment]
@yuyez can you review this PR?

**Comment by yuyez** [comment]
@tabrezm Could you please take a look? The suggested changes differ from the element docs.

**Comment by TianoMS** [comment]
@tabrezm could you review this change?

**Comment by tabrezm** [comment]
@yuyez I'm OK updating the docs to match, as long as this behavior is functionally supported by CreateUiDef. Can you confirm?

File: schemas/0.0.1-preview/CreateUIDefinition.ProviderControl.json [hunk 1]
```diff
@@ -170,7 +170,7 @@
           "items": {
             "type": "string"
           },
-          "minItems": 3
+          "minItems": 1
```
Comment (tabrezm) [inline schemas/0.0.1-preview/CreateUIDefinition.ProviderControl.json]: Could we make it 0? Recommendations shouldn't be mandatory.

File: schemas/0.0.1-preview/CreateUIDefinition.CommonControl.json [hunk 2]
```diff
@@ -459,7 +459,13 @@
           "properties": {
             "required": {
               "type": [ "boolean", "string" ]
-            }
+            },
```
Comment (tabrezm) [inline schemas/0.0.1-preview/CreateUIDefinition.CommonControl.json]: Is this actually supported properly for other elements? I"m worried that there may be at least one element where this behavior won't work as expected if we push this schema change.

Comment (yuyez) [inline schemas/0.0.1-preview/CreateUIDefinition.CommonControl.json]: This schema change only affects UserNameTextBox which I think supports regex and validationMessage.

File: schemas/0.0.1-preview/CreateUIDefinition.ProviderControl.json (hunk 1 above)
Comment (yuyez) [inline schemas/0.0.1-preview/CreateUIDefinition.ProviderControl.json]: I saw our code checking for at least one.

File: schemas/0.0.1-preview/CreateUIDefinition.CommonControl.json (hunk 2 above)
Comment (tabrezm) [inline schemas/0.0.1-preview/CreateUIDefinition.CommonControl.json]: Oops, sorry. Missed this when scrolling up in the PR. LGTM

File: schemas/0.0.1-preview/CreateUIDefinition.ProviderControl.json (hunk 1 above)
Comment (tabrezm) [inline schemas/0.0.1-preview/CreateUIDefinition.ProviderControl.json]: OK, let's bring this down to 1 in the schema; we can do a code change to support 0 if users request it.

**Comment by tabrezm** [comment]
@TianoMS feel free to merge this. I'll update the wiki once this PR is live.

**Pull request merged**